### PR TITLE
[api] Route integration events to listening jobs

### PR DESCRIPTION
**Pull request by noe-charmet**
Route bounded integration events directly to active workflow listener jobs.

Listening jobs previously depended on retired listener pseudo-events and did not have durable, per-job event routing from trigger dispatch. This adds active listener subscription projection in triggers, workflows-side listener event buffering, and dispatch behavior that treats listener-only matches as routed while preserving retry semantics for transient delivery failures.

The implementation keeps trigger definition subscriptions and job listener subscriptions separate. Active listener jobs project local subscriptions from `workflows.job.activated`, terminal jobs clean them up from `workflows.job.terminated`, and matching integration events are delivered idempotently into workflows storage.

Reviewers should look closely at dispatch outcome semantics, the folded Drizzle baseline changes, and the idempotency boundaries around `event_ref` and listener matcher ordinals.

Closes ENG-672

---
## Summary by cubic
Route integration events directly to active listening jobs using per‑job subscriptions and buffered delivery, and only mark events as routed when a job accepts them. Implements ENG-672 by removing listener pseudo‑events and enforcing valid listening activation.

- **New Features**
  - Persist active listener subscriptions from `workflows.job.activated`; prune stale ordinals on re‑projection; remove on `workflows.job.terminated`.
  - Buffer matched listener events per job in workflows with per‑job `event_ref` idempotency; lock the job row to avoid races; skip terminal or missing jobs; record metrics.
  - Update dispatcher to fan out to both definitions and listeners; listener‑only matches don’t create runs; idempotent replays succeed; skipped stale listeners now keep the event outcome as `discarded`.
  - Add `routeEventToJobListeners` API and tests; add `WORKFLOWS_JOB_ACTIVATED` in `@shipfox/api-workflows-dto` with non‑empty `on` for listening jobs; retire `workflows.listener.*` and `workflows.job_event.delivered`.

- **Migration**
  - Run DB migrations for triggers and workflows.
  - Ensure workflows emit `workflows.job.activated` (with `on`/`until`) and `workflows.job.terminated`; stop emitting retired listener events.
  - Install deps and build: `@shipfox/api-triggers` now depends on `@shipfox/api-workflows-dto`.

<sup>Written for commit a66eaf8cf7239048c646c8e9b7fd45201209e294. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/ShipfoxHQ/shipfox/pull/472?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: libs/api/triggers/drizzle/0000_initial.sql [hunk 1]
```diff
@@ -12,6 +13,19 @@ CREATE TABLE "triggers_decisions" (
 	"created_at" timestamp with time zone DEFAULT now() NOT NULL
 );
 --> statement-breakpoint
+CREATE TABLE "triggers_job_listener_subscriptions" (
```
[the remark this answers is not in the document]
Comment (noe-charmet) [reply libs/api/triggers/drizzle/0000_initial.sql:16]: Not changing this one: this branch is still pre-deploy, so the folded triggers baseline is intentional. There are no previously deployed triggers databases that need a forward migration for this schema addition.

File: libs/api/workflows/drizzle/0000_initial.sql [hunk 2]
```diff
@@ -1,13 +1,45 @@
...
+	"version" integer DEFAULT 1 NOT NULL,
+	"created_at" timestamp with time zone DEFAULT now() NOT NULL,
+	"updated_at" timestamp with time zone DEFAULT now() NOT NULL,
+	"queued_at" timestamp with time zone,
+	"started_at" timestamp with time zone,
+	"finished_at" timestamp with time zone,
+	"timed_out_at" timestamp with time zone
+);
+--> statement-breakpoint
+CREATE TABLE "workflows_job_listener_events" (
```
[the remark this answers is not in the document]
Comment (noe-charmet) [reply libs/api/workflows/drizzle/0000_initial.sql:29]: Not changing this one: this branch is still pre-deploy, so the folded workflows baseline is intentional. There are no previously deployed workflows databases that need a forward migration for this schema addition.
